### ISSUE TITLE
New 2024.11.4 version error by setup (not running anymore (previous version was running but not updating)

### ISSUE THREAD

**Issue by Rotil66** [pantherale0/ha-fuelprices#30]
### System Health details

I cannot find this info

### Checklist

- [X] I have enabled debug logging for my installation.
- [X] I have filled out the issue template to the best of my ability.
- [X] This issue only contains 1 issue (if you have multiple issues, open one issue for each issue).
- [X] This issue is not a duplicate issue of any [previous issues](https://github.com/pantherale0/ha-fuelprices/issues?q=is%3Aissue+label%3A%22Bug%22+)..

### Describe the issue

The integration is not setiting up any more after the update to 2024.11.4  the previous version was running but not updating the fuel prices without a restart. Also after re-download, reinstall, reconfigure, restart. In the integration pace the intergration is visible. (red and sayong fail to setup) The local icon is yellow the Web icon is visible (gray - internet acces is available)

### Reproduction steps

1. Permantent failing 'so reproducable'
2.
3.
...

### Debug logs

```text
In the log's I see this message:
DirectLease appears to have blocked your IP. Your options are:1) Contact tankservice-block@app-it-up.com for further support. 2) Change your WAN IP address or use a VPN. 3) Use a proxy server to connect.

But Why should it being blocked only by updating the integration. Accessing Directlease from a Laptop thru the same internet line is working OK (should be comming from the same IP address)
```

### Diagnostics dump

[home-assistant_fuel_prices_2024-11-27T12-39-24.413Z.log](https://github.com/user-attachments/files/17934323/home-assistant_fuel_prices_2024-11-27T12-39-24.413Z.log)

**Comment by Rotil66** [comment]
![image](https://github.com/user-attachments/assets/090b69d7-acbf-429f-b64d-eda7310abafb)

This is what is is also showing

**Comment by pantherale0** [comment]
Yes, the provider has blocked the connection because you polled the source too frequently.

It was raised on the forums, unfortunately there is no fix for this.

**Comment by pantherale0** [comment]
I did mention here that I do not recommend using the "workaround" for a previous issue.

https://github.com/pantherale0/ha-fuelprices/issues/25#issuecomment-2489138122

**Comment by Rotil66** [comment]
I allready disabled the workaround and the intergration was still working. Only after the update it stopped. But if blokked, I have no idea how to get my Ip deblocked. (send a mail to the mentioned address, but no reaction....yet)

⁣BlueMail voor Android downloaden ​

Op 27 nov 2024 15:04, om 15:04, Jordan Harvey ***@***.***> schreef:

**Comment by Rotil66** [comment]
Maybe we are a bit lucky. I understand that the app is build build by 'App it up BV' They happen tonbe located 10 km from my home. So I send the a direct mail with the question what I can do to solve this issue.

Let's what will happen......

Thanks,
Rolof

⁣BlueMail voor Android downloaden ​

Op 27 nov 2024 15:34, om 15:34, Jordan Harvey ***@***.***> schreef:

**Comment by Rotil66** [comment]
Hi Jordan,

I just git the very customer friendly reaction that directlease will do everything in their power to prevent people to retrieve data in any kind of way outside their app in what they say is illegal. So unfortunately I have deleted the intergration because I cannot use it anymore. Also don't want to if thus is the reply from a company.
A pitty but I will look for another way.
Of course this is nothing to blame you, because you can't do anything about this.

Kind regards,
Rolof

⁣BlueMail voor Android downloaden ​

Op 27 nov 2024 15:34, om 15:34, Jordan Harvey ***@***.***> schreef:

**Comment by pantherale0** [comment]
I've actually started a conversation with them regarding this and they are open to helping us.

Leave it with me, I'll work with directlease so we can all get what we want.

**Comment by murnty** [comment]
> I will see if they will respond and let you know. But I don't expect much from them. Is there another provider I couldnuse in the nederlands?? ⁣BlueMail voor Android downloaden ​Op 27 nov 2024 15:34, om 15:34, Jordan Harvey ***@***.***> schreef:
> […](#)

Maybe ANWB fuel price platform can be used as a source of data instead of DirectLease

**Comment by pantherale0** [comment]
> Maybe ANWB fuel price platform can be used as a source of data instead of DirectLease

Thanks for the info, I'll see what I can come up with DirectLease (or 'App It Up') first.

Understandably they don't like us accessing how we're currently accessing it because there is a cost associated with getting the data.

```
Nevertheless we appreciate your work for the community and like to discuss how we can find a better way to support the HA tweakers with our data.
```

**Comment by Rotil66** [comment]
Great,
I will wait if there will be a solution in the future.
Thanks for the great work.

Kind Regards !

⁣BlueMail voor Android downloaden ​

Op 29 nov 2024 20:48, om 20:48, Jordan Harvey ***@***.***> schreef:

**Comment by riclun80** [comment]
Is there a possibility to get the data from:
https://www.zuinigtanken.nl/tankstation-in-de-buurt/

I have it in a Webpage Card for my area for quick viewing.

**Comment by pantherale0** [comment]
Hi,

No, the data source isn't overly complete (a few random stations I've selected don't return a price).

Anyway, DirectLease will pick this up in the new year with us. There are some changes I need to implement first.

**Comment by pantherale0** [comment]
Update on DirectLease: AppItUp (the company that operates the service) is currently quite busy, however, hopefully they can pick this up during this first quarter.

**Comment by pantherale0** [comment]
No problem, we're probably going to see some licensing changes, but I did send off some suggestions for what we can do so that everyone is happy :)

**Comment by Rotil66** [comment]
I hope you can get somewhere. My IP is still blocked. Hope that can be lifted

Thanks

⁣BlueMail voor Android downloaden ​

Op 28 jan 2025 19:17, om 19:17, Jordan Harvey ***@***.***> schreef:

**Comment by snigehere** [comment]
Hi @pantherale0 .. many thanks for your fast response and assistance.  Its much appreciated.

**Comment by pantherale0** [comment]
2025.2.0 will re-enable directlease, by default it will be disabled unless manually added in the config wizard.

The next release will change the user agent so that Home Assistant traffic can be easily identified (and perhaps get these IPs unblocked).

Unfortunately still limited though, you will not be able to request data more frequently than 1 day (and a random number of hours / minutes up to 4 hours), you should also refrain from making too many requests to the find fuel station / fuel price services because these will also bump up the number of requests to the API..

I haven't heard anything further from App It Up. But will keep this updated as and when.

Now just a legal note:

For anyone reading this who is wanting to use this in a commercial setting, see the license here https://github.com/pantherale0/pyfuelprices/blob/main/pyfuelprices/sources/netherlands/LICENSE.md.. In short - data provided by DirectLease within the Python is for personal usage only within a personal Home Assistant installation and any form of commercial usage of the API that is providing this data is prohibited.

**Comment by Rotil66** [comment]
Hi @pantherale0 
It's been a while that I was looking if the fuel prices intergration is working again. (for me)
I see that there are things happening. So I reinstalled the intergration. Althought the previous error is nit seen I Unfortunately cannot setup the intergration. As soon as I try to setup an area I get an unknow error.
(I also can't select the fuel type (no selection list visible and I cannot type anything.
I deleted the whole fuel price folder and reinstalled but still it find my previous area setting. So if I use these and directlease as anprovider is activeds the intergration but the are no entity's
Any idea what's going wrong or are things not ready yet.

Thanks!!

**Comment by pantherale0** [comment]
Fuel type will populate when some entities are in HA.

What is produced in the logs?

**Comment by pantherale0** [comment]
Hi @Rot

The connection is still being blocked, just some recent changes mean that it doesn't stop the integration from starting up anymore (there were some issues with UK data sources breaking the integration lately where providers either no longer provide data or have moved the feeds).

In theory it should continue to retry while it's setup in the background. Might be worth trying to contact their support again to see if they will be ok to unblock your IP.

**Comment by pantherale0** [comment]
Hi all,

Update from me, I haven't forgotten about this... but its been radio silent from DirectLease/App It Up since my last update :(

Hopefully with the addition of Finelly as a data source, companies might start to be a bit more open...
